### DISCUSSION TITLE
Bug im Abfrage Interval?

### DISCUSSION THREAD

**Discussion by olomberg** [tbnobody/OpenDTU#472] [Q&A]
Hallo Zusammen.

Keine Ahnung, ob es sich dabei um einen Bug handelt oder ein Verständnisproblem meinerseits. Unter den DTU Einstellungen steht das Abfrage Interval auf 5 Sekunden. Habe ich 1 Wechselrichter unter den Wechselrichter Einstellungen konfiguriert, werden die Daten auch im 5 Sekunden Zyklus aktualisiert, sowie per MQTT an den Broker geschickt. In meinem Fall habe ich aber 2 Wechselrichter (meinen + der vom Nachbarn) unter den Wechselrichter Einstellungen konfiguriert. Das Interval steht nach wie vor auf 5 Sekunden. Was jetzt aber passiert ist, dass die Daten alle 10 Sekunden aktualisiert, als auch per MQTT geschickt werden (für beide Wechselrichter). Keine Ahnung wie es sich dann mit 3 Wechselrichtern oder mehr verhalten würden. Werden es dann 15,20,25,etc. Sekunden??? Habe in den Discussions und der Doku leider nichts dazu gefunden. Keine Ahnung, ob es sich um einen Bug handelt oder "by design" so ist. Falls letzteres der Fall sein sollte, wäre ein Hinweis zu dem Parameter in der Doku sehr nett.

Gruss

Oliver

**Comment by tbnobody** [comment]
> Werden es dann 15,20,25,etc. Sekunden???

Ja... Alle ${PollInterval} Sekunden werden Abfragepakete in die TX Queue eingefügt und zwar immer nur eines. Hierbei werden die Inverter durchiteriert. Da es ein Interval für die Queue, für den Websocket und für das MQTT publishen gibt kommt es zu diesen 10Sek. Werten. 

> Falls letzteres der Fall sein sollte, wäre ein Hinweis zu dem Parameter in der Doku sehr nett.

Schicke mir doch einfach einen Diff oder PR wie du dir das vorstellst.

**Reply by koerri** [reply]
How about dividing ${PollInterval} by the number of inverters, that would leave the per-inverter interval basically unchanged?
Maybe with a little increase in ${PollInterval} if there are so many inverters that the inverter-to-inverter interval becomes too short.

During normal operation longer intervals shouldn't be much of a problem, but when setting up your PV installation or for troubleshooting in general it is really helpful to get frequent updates.
